src/coreclr/vm/threads.cpp

```diff
@@ -1857,90 +1876,88 @@ BOOL Thread::HasStarted(BOOL bRequiresTSL)
     }
     EX_END_CATCH(SwallowAllExceptions);
 
-FAILURE:
```

Comment:
I broke apart the `if`/`else` and made a always failure `goto`. The contents of the two branches are largely the same apart from the locking boolean in the failure case &ndash; which is now the `TSNC_TSLTakenForStartup` flag.